### PR TITLE
Add guide for setting up Python environments with pyenv and Poetry

### LINKED ISSUE #15: Feedback link in header broken, goes to white page
The header Feedback icon (with the envelope) on http://www.berkeleybop.org/index.html hangs on a white page, probably not what it was meant to do.

>javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B

### LINKED ISSUE #70: Add more content to best practices page about Python versions and virtual environments
Currently the [best practices page](https://berkeleybop.github.io/best_practice#codingpython) make a brief statements that projects should use Python 3.8+ and use `poetry` to manage virtual environments. It seems to be a semi-regular source of confusion as to _how_ to actually accomplish that. I think we should expand that section or add a new one that describes -- starting from scratch -- how to install and use `pyenv` to manage Python versions and how to install and use `poetry` to manage virtual environments. We shouldn't rewrite the entire documentation of those projects, but rather just outline the basic commands needed to get started with a new or existing project.

### PR DESCRIPTION
**Pull request by pkalita-lbl**
These changes add a new page aimed at new-ish developers to get them on the right track using pyenv and Poetry, as well as ensuring best practices are made explicit even for existing developers. I'm trying to balance providing concise, "just run these commands" instructions with providing context about why they need to be run. I'm a little concerned that what I have here to a bit too wordy, but maybe it's okay.

Fixes #70 
Fixes #15 

cc: @matentzn @sierra-moxon

### CODE REVIEW

**Review by sierra-moxon — APPROVED** [review]

**Review by nlharris — APPROVED** [review]
Thanks for putting this together! Let's merge it and see how it looks and then possibly I'll have minor suggestions to make (via PR).
I'll also test the feedback link once it's merged.
